File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/RecursosHumanos/CategoriaRHForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoriaRHForm {
    LocalSap: StringEditor;
    Plantilla: StringEditor;
    Vacantes: StringEditor;
}

export class CategoriaRHForm extends PrefixedContext {
    static formKey = 'RecursosHumanos.CategoriaRH';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriaRHForm.init)  {
            CategoriaRHForm.init = true;

            var w0 = StringEditor;

            initFormType(CategoriaRHForm, [
                'LocalSap', w0,
                'Plantilla', w0,
                'Vacantes', w0
            ]);
        }
    }
}

File: MasterDirectory/MasterDirectory.Web/Modules/RecursosHumanos/CategoriaRH/CategoriaRHForm.cs
using Serenity.ComponentModel;
using System;

namespace MasterDirectory.RecursosHumanos.Forms;

[FormScript("RecursosHumanos.CategoriaRH")]
[BasedOnRow(typeof(CategoriaRHRow), CheckNames = true)]
public class CategoriaRHForm
{
    [Tab("General")]
    public string LocalSap { get; set; }
    public string Plantilla { get; set; }
    public string Vacantes { get; set; }
    //public DateTime DtRegistro { get; set; }
}